extensions/amp-story/1.0/amp-story-page-attachment.js

```diff
@@ -107,29 +108,36 @@ export class AmpStoryPageAttachment extends DraggableDrawer {
    * @private
    */
   buildInline_() {
-    const closeButtonEl = this.headerEl_.appendChild(
-      htmlFor(this.element)`
+    const closeButtonEl = htmlFor(this.element)`
           <button class="i-amphtml-story-page-attachment-close-button" aria-label="close"
               role="button">
-          </button>`
-    );
+          </button>`;
     const localizationService = getLocalizationService(devAssert(this.element));
+
+    const titleEl = htmlFor(this.element)`
+    <span class="i-amphtml-story-page-attachment-title"></span>`;
+
     if (localizationService) {
       const localizedCloseString = localizationService.getLocalizedString(
         LocalizedStringId.AMP_STORY_CLOSE_BUTTON_LABEL
       );
       closeButtonEl.setAttribute('aria-label', localizedCloseString);
     }
 
-    this.headerEl_.appendChild(
-      htmlFor(this.element)`
-          <span class="i-amphtml-story-page-attachment-title"></span>`
-    );
-
     if (this.element.hasAttribute('data-title')) {
-      this.headerEl_.querySelector(
-        '.i-amphtml-story-page-attachment-title'
-      ).textContent = this.element.getAttribute('data-title');
+      titleEl.textContent = this.element.getAttribute('data-title');
+    }
+
+    if (isPageAttachmentUiV2ExperimentOn(this.win)) {
+      const titleAndCloseWrapperEl = this.headerEl_.appendChild(
+        htmlFor(this.element)`
+            <div class="i-amphtml-story-draggable-drawer-header-title-and-close"></div>`
+      );
+      titleAndCloseWrapperEl.appendChild(closeButtonEl);
+      titleAndCloseWrapperEl.appendChild(titleEl);
+    } else {
+      this.headerEl_.appendChild(closeButtonEl);
+      this.headerEl_.appendChild(titleEl);
```

Comment:
This seems repetitive (adding the close button and title to an element in both branches of the `if` statement), plus we're creating 3 HTML templates that will mostly always have the same DOM structure. Would it be possible to simplify the logic here to always have the `title-and-close` div inside the header, and instead of 3 templates, to make 1 larger one (you can use `htmlRefs` to get the elements easily)?

Reply:
It only needs the wrapper in the V2 experiment.
I suppose we could add the wrapper but the intention is to not modify the current implementation during the experiment.